### PR TITLE
IOS DB plugin implementation

### PR DESCRIPTION
**Pull request by Mani397**

### CODE REVIEW

File: plugin.xml [hunk 1]
```diff
@@ -26,7 +26,7 @@
       <source-file src="src/android/org/sunbird/db/SunbirdDBContext.java" target-dir="src/org/sunbird/db" />
       <source-file src="src/android/org/sunbird/db/SunbirdDBHelper.java" target-dir="src/org/sunbird/db" />
       <source-file src="src/android/org/sunbird/db/SunbirdDBPlugin.java" target-dir="src/org/sunbird/db" />
```
Comment (sharathkashyap) [inline plugin.xml:28]: Was this change necessary

File: src/ios/DBContext.swift [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+class DBContext {
+    static var name = ""
+    static var version = 3
+    private static var db: OpaquePointer? = nil
+    private static var dbHelper = DBHelper()
+
+    static func getOperator() -> OpaquePointer? {
+        let databaseFilePath = "\(NSSearchPathForDirectoriesInDomains(.documentDirectory, .userDomainMask, true)[0])/\(DBContext.name)"
+            if dbHelper.createDatabaseFile(databaseFilePath) != true {
```
Comment (sharathkashyap) [inline src/ios/DBContext.swift:12]: Can you please test this case. Nil return might cause a problem.

File: src/ios/DBContext.swift [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+                return nil
+            }
+            if sqlite3_open_v2(databaseFilePath, &db, SQLITE_OPEN_READWRITE, nil) != SQLITE_OK
+            {
```
Comment (sharathkashyap) [inline src/ios/DBContext.swift:16]: Can we propogate the Custom Error onto JS so that you can interpret these errors in promises

**Review by sharathkashyap — APPROVED** [review]

File: plugin.xml (hunk 1 above)
Comment (HarishGangula) [reply plugin.xml:28]: this indentation fix

File: src/ios/DBContext.swift (hunk 2 above)
Comment (HarishGangula) [reply src/ios/DBContext.swift:12]: This is checked

**Review by swayangjit — APPROVED** [review]
